backend/FwLite/FwDataMiniLcmBridge/Api/LcmHelpers.cs

```diff
@@ -148,18 +206,42 @@ internal static int GetWritingSystemHandle(this LcmCache cache, WritingSystemId
         return multiString.get_String(wsHandle)?.Text ?? null;
     }
 
-    internal static IMoStemAllomorph CreateLexemeForm(this LcmCache cache)
+    internal static IMoForm CreateLexemeForm(this LcmCache cache, MorphType morphType)
+    {
+        return
+            IsAffixMorphType(morphType)
+            ? cache.ServiceLocator.GetInstance<IMoAffixAllomorphFactory>().Create()
+            : cache.ServiceLocator.GetInstance<IMoStemAllomorphFactory>().Create();
+    }
+
+    internal static bool IsAffixMorphType(MorphType morphType)
     {
-        return cache.ServiceLocator.GetInstance<IMoStemAllomorphFactory>().Create();
+        return morphType switch
+        {
+            // Affixes of all types should use the Affix morph type factory
+            MorphType.Circumfix => true,
+            MorphType.Infix => true,
+            MorphType.Prefix => true,
+            MorphType.Simulfix => true,
+            MorphType.Suffix => true,
+            MorphType.Suprafix => true,
+            MorphType.InfixingInterfix => true,
+            MorphType.PrefixingInterfix => true,
+            MorphType.SuffixingInterfix => true,
+
+            // Everything else should use the Stem morph type factory
+            _ => false,
+        };
     }
 
-    internal static ILexEntry CreateEntry(this LcmCache cache, Guid id)
+    internal static ILexEntry CreateEntry(this LcmCache cache, Guid id, MorphType morphType)
     {
         var lexEntry = cache.ServiceLocator.GetInstance<ILexEntryFactory>().Create(id,
             cache.ServiceLocator.GetInstance<ILangProjectRepository>().Singleton.LexDbOA);
-        lexEntry.LexemeFormOA = cache.CreateLexemeForm();
+        lexEntry.LexemeFormOA = cache.CreateLexemeForm(morphType);
         //must be done after the IMoForm is set on the LexemeForm property
-        lexEntry.LexemeFormOA.MorphTypeRA = cache.ServiceLocator.GetInstance<IMoMorphTypeRepository>().GetObject(MoMorphTypeTags.kguidMorphStem);
+        var lcmMorphType = ToLcmMorphTypeId(morphType) ?? ToLcmMorphTypeId(MorphType.Stem);
+        lexEntry.LexemeFormOA.MorphTypeRA = cache.ServiceLocator.GetInstance<IMoMorphTypeRepository>().GetObject(lcmMorphType!.Value);
```

Comment:
Hum, I think we need a way for this to also be done in the proxy where `CreateLexemeForm` is also called. 
Maybe we make another helper: `SetLexemeForm(ILexEntry entry, MorphType type, LcmCache cache)` which wraps all this up so it can be used both here and in the proxy

Reply:
Done in commit 6b537efdd.